### ISSUE TITLE
Huawei vrpv8 commit fails to complete successfully

### ISSUE THREAD

**Issue by wayneshow** [ktbyers/netmiko#958]
script :
```
from netmiko import ConnectHandler
huawei = {'device_type': 'huawei_vrpv8', 'ip': 'x.x.x.x', 'username': 'guaziadmin', 'password': 'password'}
net_connect = ConnectHandler(**huawei)

commands = ['vlan 10']

output1 = net_connect.send_config_set(commands)
print(output1)
```

error:
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/paramiko/channel.py", line 683, in recv
    out = self.in_buffer.read(nbytes, self.timeout)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/paramiko/buffered_pipe.py", line 160, in read
    raise PipeTimeout()
paramiko.buffered_pipe.PipeTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/base_connection.py", line 448, in _read_channel_expect
    new_data = self.remote_conn.recv(MAX_BUFFER)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/paramiko/channel.py", line 685, in recv
    raise socket.timeout()
socket.timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sunshouwen/PycharmProjects/test1/netssh_test.py", line 7, in <module>
    output1 = net_connect.send_config_set(commands)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/base_connection.py", line 1402, in send_config_set
    output += self.exit_config_mode()
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/huawei/huawei_ssh.py", line 26, in exit_config_mode
    return super(HuaweiSSH, self).exit_config_mode(exit_config=exit_config)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/cisco_base_connection.py", line 47, in exit_config_mode
    pattern=pattern)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/base_connection.py", line 1324, in exit_config_mode
    output = self.read_until_pattern(pattern=pattern)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/base_connection.py", line 520, in read_until_pattern
    return self._read_channel_expect(*args, **kwargs)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/netmiko/base_connection.py", line 456, in _read_channel_expect
    raise NetMikoTimeoutException("Timed-out reading channel, data not available.")
netmiko.ssh_exception.NetMikoTimeoutException: Timed-out reading channel, data not available.
```

**Comment by smk4664** [comment]
If you're not on the develop branch already, I would suggest switching to it and testing your script. There was a recent commit to correct the send_config_set for huawei.

**Comment by ktbyers** [comment]
Agreed, can you re-test this using the `develop` branch and see if the issue still exists.

**Comment by wayneshow** [comment]
> If you're not on the develop branch already, I would suggest switching to it and testing your script. There was a recent commit to correct the send_config_set for huawei.

when I switched develop branch, either display commands or set commands generate the same error 

`
Traceback (most recent call last):
  File "/Users/sunshouwen/PycharmProjects/test1/netssh_test.py", line 5, in <module>
    net_connect = ConnectHandler(**huawei)
  File "./netmiko/netmiko/ssh_dispatcher.py", line 215, in ConnectHandler
    return ConnectionClass(*args, **kwargs)
  File "./netmiko/netmiko/base_connection.py", line 270, in __init__
    self.establish_connection()
  File "./netmiko/netmiko/base_connection.py", line 769, in establish_connection
    self.remote_conn_pre.connect(**ssh_connect_params)
TypeError: connect() got an unexpected keyword argument 'passphrase'

`

**Comment by wayneshow** [comment]
> If you're not on the develop branch already, I would suggest switching to it and testing your script. There was a recent commit to correct the send_config_set for huawei.

switch develop branch , have the same issue.....

**Comment by ktbyers** [comment]
@wayneshow So it errors on `exit_config_mode` after trying to do your configuration commands.

Can you show what doing the above changes (config changes) looks like if you do them manually?

**Comment by wayneshow** [comment]
I switched develop branch, and did the same commands. 

`sunshouwen@sunshouwendeMacBook-Pro  ~/PycharmProjects/test1/netmiko   develop  git branch
* develop`
-------------------------------
import sys
sys.path.append('./netmiko')
from netmiko import ConnectHandler
huawei = {'device_type': 'huawei_vrpv8', 'ip': 'x.x.x.x', 'username': 'admin', 'password': 'password'}
net_connect = ConnectHandler(**huawei)

commands = ['vlan 10']

output1 = net_connect.send_config_set(commands)
print(output1)
------------errors-------------
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/paramiko/channel.py", line 699, in recv
    out = self.in_buffer.read(nbytes, self.timeout)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/paramiko/buffered_pipe.py", line 164, in read
    raise PipeTimeout()
paramiko.buffered_pipe.PipeTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./netmiko/netmiko/base_connection.py", line 470, in _read_channel_expect
    new_data = self.remote_conn.recv(MAX_BUFFER)
  File "/usr/local/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/paramiko/channel.py", line 701, in recv
    raise socket.timeout()
socket.timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sunshouwen/PycharmProjects/test1/netssh_test.py", line 9, in <module>
    output1 = net_connect.send_config_set(commands)
  File "./netmiko/netmiko/base_connection.py", line 1427, in send_config_set
    output += self.exit_config_mode()
  File "./netmiko/netmiko/huawei/huawei_ssh.py", line 27, in exit_config_mode
    pattern=pattern)
  File "./netmiko/netmiko/cisco_base_connection.py", line 47, in exit_config_mode
    pattern=pattern)
  File "./netmiko/netmiko/base_connection.py", line 1349, in exit_config_mode
    output = self.read_until_pattern(pattern=pattern)
  File "./netmiko/netmiko/base_connection.py", line 542, in read_until_pattern
    return self._read_channel_expect(*args, **kwargs)
  File "./netmiko/netmiko/base_connection.py", line 478, in _read_channel_expect
    raise NetMikoTimeoutException("Timed-out reading channel, data not available.")
netmiko.ssh_exception.NetMikoTimeoutException: Timed-out reading channel, data not available.




Same error as before！！！！

**Comment by ktbyers** [comment]
@wayneshow Yes, I need to see what these configuration commands look like when you do them manually in the CLI.

Can you copy-and-paste, what making these config changes looks like in the CLI, including doing a commit (if necessary) and also exiting config mode)?

**Comment by wayneshow** [comment]
@ktbyers 
copy from CLI:  (I have hiddened password and IP)

`
>>> import sys
>>> sys.path.append("/Users/sunshouwen/PycharmProjects/test1/netmiko")
>>> from netmiko import ConnectHandler
>>> huawei = {'device_type': 'huawei_vrpv8', 'ip': 'x.x.x.x', 'username': 'admin', 'password': 'password'}
>>> net_connect = ConnectHandler(**huawei)
>>> commands = ['vlan 10']
>>> output1 = net_connect.send_config_set(commands)
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/paramiko/channel.py", line 699, in recv
    out = self.in_buffer.read(nbytes, self.timeout)
  File "/usr/local/lib/python3.6/site-packages/paramiko/buffered_pipe.py", line 164, in read
    raise PipeTimeout()
paramiko.buffered_pipe.PipeTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/base_connection.py", line 470, in _read_channel_expect
    new_data = self.remote_conn.recv(MAX_BUFFER)
  File "/usr/local/lib/python3.6/site-packages/paramiko/channel.py", line 701, in recv
    raise socket.timeout()
socket.timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/base_connection.py", line 1427, in send_config_set
    output += self.exit_config_mode()
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/huawei/huawei_ssh.py", line 27, in exit_config_mode
    pattern=pattern)
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/cisco_base_connection.py", line 47, in exit_config_mode
    pattern=pattern)
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/base_connection.py", line 1349, in exit_config_mode
    output = self.read_until_pattern(pattern=pattern)
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/base_connection.py", line 542, in read_until_pattern
    return self._read_channel_expect(*args, **kwargs)
  File "/Users/sunshouwen/PycharmProjects/test1/netmiko/netmiko/base_connection.py", line 478, in _read_channel_expect
    raise NetMikoTimeoutException("Timed-out reading channel, data not available.")
netmiko.ssh_exception.NetMikoTimeoutException: Timed-out reading channel, data not available.
>>>
`

**Comment by ktbyers** [comment]
No, not from Python; the CLI from the Huawei device. 

I.E. if you manually try to do your config changes on the Huawei device. Can you show me what that looks like.

**Comment by ktbyers** [comment]
Okay, try:

```
output1 = net_connect.send_config_set(commands, exit_config_mode=False)
net_connect.commit()
```

Make sure commands is only 'vlan 10'.


Kirk

**Comment by wayneshow** [comment]
I execute the script after execute the config above , there is no error .  In huawei CLI , I execute "dis  history-command all", saw  execute return without executing commit before. 

`output1 = net_connect.send_config_set(commands, exit_config_mode=False)
net_connect.commit()`

Traceback (most recent call last):
  File "/Users/sunshouwen/PycharmProjects/test1/netssh_test.py", line 10, in <module>
    net_connect.commit()
  File "./netmiko/netmiko/huawei/huawei_ssh.py", line 114, in commit
    strip_command=False, delay_factor=delay_factor)
  File "./netmiko/netmiko/base_connection.py", line 1193, in send_command_expect
    return self.send_command(*args, **kwargs)
  File "./netmiko/netmiko/base_connection.py", line 1175, in send_command
    search_pattern))
OSError: Search pattern never detected in send_command_expect: \[\*BMY\.C102\.F09\-Leaf\-CE6855\-1\-vlan12\]

**Comment by ktbyers** [comment]
Okay, there is a bug in `commit`

How long does commit take to execute (roughly) when you run it manually.

**Comment by carlniger** [comment]
I'm going to close this since it looks like it maybe was fixed in #1073 and its pretty old. Feel free to reopen if this is still an issue on the current develop branch.

**Comment by nontha** [comment]
`output1 = net_connect.send_config_set(commands, exit_config_mode=False) `

This extended the number of times a loop run in my script. But didn't solve the issue completely.
